integration/spark-common-test/src/test/scala/org/apache/carbondata/integration/spark/testsuite/timeseries/TestTimeseriesTableSelection.scala

```diff
@@ -99,124 +113,763 @@ class TestTimeseriesTableSelection extends QueryTest with BeforeAndAfterAll {
+    val df = sql(
+      """
+        | SELECT
+        |   TIMESERIES(mytime,'year') AS yearLevel,
+        |   TIMESERIES(mytime,'month') AS monthLevel,
+        |   TIMESERIES(mytime,'day') AS dayLevel,
+        |   TIMESERIES(mytime,'hour') AS hourLevel,
+        |   TIMESERIES(mytime,'minute') AS minuteLevel,
+        |   TIMESERIES(mytime,'second') AS secondLevel,
+        |   SUM(age) AS SUM
+        | FROM mainTable
+        | WHERE TIMESERIES(mytime,'minute')='2016-02-23 09:01:00'
+        | GROUP BY
+        |   TIMESERIES(mytime,'year'),
+        |   TIMESERIES(mytime,'month'),
+        |   TIMESERIES(mytime,'day'),
+        |   TIMESERIES(mytime,'hour'),
+        |   TIMESERIES(mytime,'minute'),
+        |   TIMESERIES(mytime,'second')
+        | ORDER BY
+        |   TIMESERIES(mytime,'year'),
+        |   TIMESERIES(mytime,'month'),
+        |   TIMESERIES(mytime,'day'),
+        |   TIMESERIES(mytime,'hour'),
+        |   TIMESERIES(mytime,'minute'),
+        |   TIMESERIES(mytime,'second')
+      """.stripMargin)
+
+    checkExistence(df, true,
+      "2016-01-01 00:00:00",
+      "2016-02-01 00:00:00",
+      "2016-02-23 09:00:00",
+      "2016-02-23 09:01:00",
+      "2016-02-23 09:01:50",
+      "30"
+    )
+  }
+
+  test("test timeseries table selection 19: select with many GROUP BY AND many filter") {
+    val df = sql(
+      """
+        | SELECT
+        |   TIMESERIES(mytime,'year') AS yearLevel,
+        |   TIMESERIES(mytime,'month') AS monthLevel,
+        |   TIMESERIES(mytime,'day') AS dayLevel,
+        |   TIMESERIES(mytime,'hour') AS hourLevel,
+        |   TIMESERIES(mytime,'minute') AS minuteLevel,
+        |   TIMESERIES(mytime,'second') AS secondLevel,
+        |   SUM(age) AS SUM
+        | FROM mainTable
+        | WHERE
+        |   TIMESERIES(mytime,'second')='2016-02-23 09:01:50' and
+        |   TIMESERIES(mytime,'minute')='2016-02-23 09:01:00' and
+        |   TIMESERIES(mytime,'hour')='2016-02-23 09:00:00' and
+        |   TIMESERIES(mytime,'month')='2016-02-01 00:00:00' and
+        |   TIMESERIES(mytime,'year')='2016-01-01 00:00:00'
+        | GROUP BY
+        |   TIMESERIES(mytime,'year'),
+        |   TIMESERIES(mytime,'month'),
+        |   TIMESERIES(mytime,'day'),
+        |   TIMESERIES(mytime,'hour'),
+        |   TIMESERIES(mytime,'minute'),
+        |   TIMESERIES(mytime,'second')
+        | ORDER BY
+        |   TIMESERIES(mytime,'year'),
+        |   TIMESERIES(mytime,'month'),
+        |   TIMESERIES(mytime,'day'),
+        |   TIMESERIES(mytime,'hour'),
+        |   TIMESERIES(mytime,'minute'),
+        |   TIMESERIES(mytime,'second')
+      """.stripMargin)
+
+    checkExistence(df, true,
+      "2016-01-01 00:00:00",
+      "2016-02-01 00:00:00",
+      "2016-02-23 09:00:00",
+      "2016-02-23 09:01:00",
+      "2016-02-23 09:01:50",
+      "30"
+    )
+  }
+
+  test("test timeseries table selection 20: filter < AND >") {
```

Comment:
Filter scenarios are already covered in TestTimeSeriesMatchStrategySuite.scala. Please remove duplicate scenarios

Reply:
different purpose, but I can remove this.